### PR TITLE
Propagate openshift detection result

### PR DESCRIPTION
**Pull request by lulf**

### CODE REVIEW

**Comment by lulf** [comment]
@enmasse-ci run tests profile=smoke crc

File: pkg/controller/address_space_controller/controller.go [hunk 1]
```diff
@@ -205,6 +205,9 @@ func ApplyDeployment(deployment *appsv1.Deployment) error {
 		applyImageEnv(container, "TOPIC_FORWARDER_IMAGE", "topic-forwarder")
 		applyImageEnv(container, "MQTT_GATEWAY_IMAGE", "mqtt-gateway")
 		applyImageEnv(container, "MQTT_LWT_IMAGE", "mqtt-lwt")
+		if util.IsOpenshift() {
+			applyImageEnv(container, "ENMASSE_OPENSHIFT", "true")
```
Comment (k-wall) [inline pkg/controller/address_space_controller/controller.go:209]:
```suggestion
			applyImageEnv(container, util.EnMasseOpenshiftEnvVar, "true")
```

**Comment by lulf** [comment]
@k-wall Even today, the operator takes a while to decide its not on openshift and produce a warning, I don't see how this is much different. It could be the case where it is in the process of attaching an aggregated API server, so a 404 could be worth retrying for.

**Comment by lulf** [comment]
@k-wall I've removed the change to the java code. I agree just changing so that the operator sets the env is sufficient, and less chances of causing issues.

**Comment by lulf** [comment]
@enmasse-ci run tests profile=isolated

**Pull request merged**